plugins/modules/azure_rm_virtualmachine_info.py

```diff
@@ -459,11 +459,16 @@ def serialize_vm(self, vm):
 
         if vm.security_profile is not None:
             new_result['security_profile'] = dict()
-            new_result['security_profile']['encryption_at_host'] = vm.security_profile.encryption_at_host
-            new_result['security_profile']['security_type'] = vm.security_profile.security_type
-            new_result['security_profile']['uefi_settings'] = dict()
-            new_result['security_profile']['uefi_settings']['secure_boot_enabled'] = vm.security_profile.uefi_settings.secure_boot_enabled
-            new_result['security_profile']['uefi_settings']['v_tpm_enabled'] = vm.security_profile.uefi_settings.v_tpm_enabled
+            if vm.security_profile.encryption_at_host is not None:
+                new_result['security_profile']['encryption_at_host'] = vm.security_profile.encryption_at_host
+            if vm.security_profile.security_type is not None:
+                new_result['security_profile']['security_type'] = vm.security_profile.security_type
+            if vm.security_profile.uefi_settings is not None:
+                new_result['security_profile']['uefi_settings'] = dict()
+                if vm.security_profile.uefi_settings.secure_boot_enabled is not None:
+                   new_result['security_profile']['uefi_settings']['secure_boot_enabled'] = vm.security_profile.uefi_settings.secure_boot_enabled
+                if vm.security_profile.uefi_settings.v_tpm_enabled is not None:
+                   new_result['security_profile']['uefi_settings']['v_tpm_enabled'] = vm.security_profile.uefi_settings.v_tpm_enabled
```

Comment:
indentation is not a multiple of 4
```suggestion
                    new_result['security_profile']['uefi_settings']['v_tpm_enabled'] = vm.security_profile.uefi_settings.v_tpm_enabled
```